docs/source/api_docs/internals.rst

```diff
@@ -0,0 +1,100 @@
+Internals
+---------
+
+This page includes information to help users understand the internal
+data structure of cuspatial.
+
+GeoArrow Format
++++++++++++++++
+
+Geospatial data is context rich - aside from just a set of
+numbers representing coordinates, they together represent certain geometry
+that requires grouping. For example, given 5 points in a plane,
+they could be 5 separate points, 2 line segments, a single linestring,
+or a pantagon. Many geometry libraries stores the points in
+array of geometric objects, commonly known as "Array of Structure" (AoS).
+AoS is not efficient for accelerated computing on parallel devices such
+as GPU. Therefore, GeoArrow format was introduced to store geodata in
+densely packed format, commonly known as "Structure of Array" (SoA).
+
+The GeoArrow format specifies a tabular data format for geometry
+information. Supported types include `Point`, `MultiPoint`, `LineString`,
+`MultiLineString`, `Polygon`, and `MultiPolygon`. In order to store
+these coordinate types in a strictly tabular fashion, columns are
+created for Points, MultiPoints, LineStrings, and Polygons.
+MultiLines and MultiPolygons are stored in the same data structure
+as LineStrings and Polygons.
+
+GeoArrow format packs complex geometry types into 14 single-column Arrow
+tables. See :func:`GeoArrowBuffers<cuspatial.GeoArrowBuffers>` docstring
+for the complete list of keys for the columns.
+
+Examples
+********
+
+The `Point` geometry is the simplest: N points are stored in a length 2*N
+buffer with interleaved x,y coordinates. An optional z buffer of length N
+can be used.
+
+The `Multipoint` geometry is the second simplest - identical to points,
+with the addition of a `multipoints_offsets` buffer. The offsets buffer
+stores N+1 indexes. The first multipoint is specified by 0, which is always
+stored in offsets[0], and offsets[1], which is the length in points of
+the first multipoint geometry. Subsequent multipoints are the prefix-sum of
+the lengths of previous multipoints.
```

Comment:
Invoking prefix sum here implies that the lengths are stored, which I don't believe they are.

```suggestion
A `Multipoint` is a group of points, and is the second simplest GeoArrow 
geometry type. It is identical to points, with the addition of a 
`multipoints_offsets` buffer. The offsets buffer stores N+1 indexes. The
 first multipoint offset is specified by 0, which is always stored in 
 `offsets[0]`. The second offset is stored in `offsets[1]`, and so on.
 The length of each multipoint `i` is the difference between `offsets[i+1]`
 and `offsets[i]`. 
```